src/plugins/telemetry/public/plugin.ts

```diff
@@ -61,6 +62,7 @@ export interface TelemetryPluginConfig {
   optInStatusUrl: string;
   sendUsageFrom: 'browser' | 'server';
   telemetryNotifyUserAboutOptInDefault?: boolean;
+  userCanChangeSettings?: boolean;
```

Comment:
I'm still concerned over the complexity of how we work out when the banner should be shown and if the user should be able to change opt-in/opt-out settings. We already have to consider 3 different settings that can be set in `kibana.yml` and then also allow for these to be changed in Advanced Settings (aside: we need to be more explicit about them in the docs. [Usage data](https://www.elastic.co/guide/en/kibana/current/advanced-options.html#kibana-telemetry-settings) doesn't give much info). It just seems overly complicated for a question of 'should we show the banner or not'.

Reply:
I agree, maybe we open a discussion to review our current approach. But I wouldn't hold this PR for that because it's fixing a bug in the logic based on the current expected behaviou. What do you think?

Reply:
I agree we should do that in a follow up PR. Let's get this bug fixed first.